pennylane/transforms/diagonalize_measurements.py

```diff
@@ -60,6 +60,14 @@ def diagonalize_measurements(tape, supported_base_obs=_default_supported_obs, to
         circuit measurements, the :func:`split_non_commuting <pennylane.transforms.split_non_commuting>` transform
         can be applied.
 
+    .. note::
+        This transform will diagonalize what it can, i.e., ``qml.X``, ``qml.Y``, ``qml.Z``,
+        ``qml.Hadamard``, ``qml.Identity``, or a linear combination of them. Any unrecognized
+        observable will not raise an error, deferring to the device's validation for supported
+        measurements later on. Lastly, if ``diagonalize_measurements`` produces additional gates
+        that the device does not support, the :func:`~pennylane.devices.preprocess.decompose``
+        transform should be applied to ensure that the additional gates are decomposed to those
```

Comment:
```suggestion
        transform is applied internally to ensure that the additional gates are decomposed to those
```

Reply:
The `decompose` transform is not automatically applied after `diagonalize_measurements`. Well it is if the device is using our default preprocessing transform, but if the device overrides its own `preprocess`, they need to remember to apply `decompose` after `diagonalize_measurements` themselves, so "should" is more appropriate here.